tests/e2e/README.md

```diff
@@ -1,4 +1,120 @@
-## Updating the trace format and tests when adjusting the framework
+# End-to-end testing
+
+This folder contains end-to-end tests and related infrastructure.
+The goal of end-to-end tests is to test our application completely.
+End-to-end in this context means tests
+where the whole app (including command line interfaces, interfaces to external components like
+relayers, CometBFT, etc) is in scope for testing.
+
+The high-level overview of what a single test case does is:
+* The test starts a docker container, see [the startup script](testnet-scripts/start-docker.sh)
+* We run a defined sequence of actions and expected states, see as an example the steps for [testing the democracy module](steps_democracy.go)
+    * Actions are things like: submitting transactions to a node, but also things like making nodes double-sign, starting a relayer, starting a new chain, ..., in short anything
+that might modify the system state meaningfully
+    * Expected states defined as parts of the system state that have to match. For example, after an action that modifies validator balances, we would specify
+    the validator balances, so those would be checked against the actual validator balances. We might not care about the governance proposals that are on chain
+    though, and just leave that part of the state unspecified.
```

Comment:
This is unclear.

Reply:
I reformulated this, could you recheck whether it is clearer?